app/js/ImageOptimiser.js

```diff
@@ -1,44 +1,41 @@
-/* eslint-disable
-    no-unused-vars,
-*/
-// TODO: This file was created by bulk-decaffeinate.
-// Fix any style issues and re-enable lint.
-/*
- * decaffeinate suggestions:
- * DS102: Remove unnecessary code created because of implicit returns
- * DS207: Consider shorter variations of null checks
- * Full docs: https://github.com/decaffeinate/decaffeinate/blob/master/docs/suggestions.md
- */
-const { exec } = require('child_process')
 const logger = require('logger-sharelatex')
-const Settings = require('settings-sharelatex')
+const metrics = require('metrics-sharelatex')
+const { callbackify } = require('util')
+const safeExec = require('./SafeExec').promises
 
 module.exports = {
-  compressPng(localPath, callback) {
-    const startTime = new Date()
-    logger.log({ localPath }, 'optimising png path')
-    const args = `optipng ${localPath}`
-    const opts = {
-      timeout: 30 * 1000,
-      killSignal: 'SIGKILL'
-    }
-    if (!Settings.enableConversions) {
-      const error = new Error('Image conversions are disabled')
-      return callback(error)
+  compressPng: callbackify(compressPng),
+  promises: {
+    compressPng
+  }
+}
+
+async function compressPng(localPath, callback) {
+  const timer = new metrics.Timer('compressPng')
+  logger.log({ localPath }, 'optimising png path')
+
+  const args = ['optipng', localPath]
+  const opts = {
+    timeout: 30 * 1000,
+    killSignal: 'SIGKILL'
+  }
+
+  try {
+    await safeExec(args, opts)
+    timer.done()
+    logger.log({ localPath }, 'finished compressing png')
+  } catch (err) {
```

Comment:
any reason for using a `try` `catch` block over doing the `.then` `.catch`?

Reply:
When using `await` we will receive the 'return' value of the function and it will throw an exception on error... Whereas if you called the method without `await` you would need to `.then` (and `.catch`) instead.

Stylistically, `async`/`await` reads more like Ruby-code than js code.

You can't say `await foo().then()` because you've already resolved the promise with `await`. (Unless `foo` returns a promise that returns a promise.)

Reply:
Technically `.then()` will aways return a promise (it's part of the contract of a promise). You can prove this to yourself by running `Promise.resolve('foo').then()` and observing that the return value is a promise.

So you can do `await foo().then()` but there's not really much point you might as well do:

```js
const resolvedValue = await foo()
// do something with resolvedValue
```